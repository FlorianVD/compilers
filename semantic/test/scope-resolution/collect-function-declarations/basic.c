// RUN-WITH-ARGS: -dump-function-table

int f1() {}

float f2() {}

float f3(int x) {}

float f4(int x, float y) {}

int f5(float x, int y, float z) {}

string f6(int x, float y, string z) {}

void f7(int x) {}
