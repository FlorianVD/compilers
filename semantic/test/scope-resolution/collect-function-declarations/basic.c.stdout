└── Program <24>
    ├── FuncDecl: returnType = 'int', name = 'f1' <1>
    │   └── CompoundStmt <0>
    ├── FuncDecl: returnType = 'float', name = 'f2' <3>
    │   └── CompoundStmt <2>
    ├── FuncDecl: returnType = 'float', name = 'f3' <6>
    │   ├── VarDecl: type = 'int', name = 'x' <4>
    │   └── CompoundStmt <5>
    ├── FuncDecl: returnType = 'float', name = 'f4' <10>
    │   ├── VarDecl: type = 'int', name = 'x' <7>
    │   ├── VarDecl: type = 'float', name = 'y' <8>
    │   └── CompoundStmt <9>
    ├── FuncDecl: returnType = 'int', name = 'f5' <15>
    │   ├── VarDecl: type = 'float', name = 'x' <11>
    │   ├── VarDecl: type = 'int', name = 'y' <12>
    │   ├── VarDecl: type = 'float', name = 'z' <13>
    │   └── CompoundStmt <14>
    ├── FuncDecl: returnType = 'string', name = 'f6' <20>
    │   ├── VarDecl: type = 'int', name = 'x' <16>
    │   ├── VarDecl: type = 'float', name = 'y' <17>
    │   ├── VarDecl: type = 'string', name = 'z' <18>
    │   └── CompoundStmt <19>
    └── FuncDecl: returnType = 'void', name = 'f7' <23>
        ├── VarDecl: type = 'int', name = 'x' <21>
        └── CompoundStmt <22>
Function table:
f1                  i64 ()
f2                  float ()
f3                  float (i64)
f4                  float (i64, float)
f5                  i64 (float, i64, float)
f6                  i8* (i64, float, i8*)
f7                  void (i64)
